aptos-move/block-executor/src/scheduler_v2.rs

```diff
@@ -66,3 +422,1400 @@ impl ExecutionQueueManager {
+            is_done: CachePadded::new(AtomicBool::new(false)),
+            is_halted: CachePadded::new(AtomicBool::new(false)),
+            queueing_commits_lock: CachePadded::new(ArmedLock::new()),
+            post_commit_processing_queue: CachePadded::new(ConcurrentQueue::<TxnIndex>::bounded(
+                num_txns as usize,
+            )),
+            committed_marker: (0..num_txns)
+                .map(|_| CachePadded::new(AtomicU8::new(CommitMarkerFlag::NotCommitted as u8)))
+                .collect(),
+        }
+    }
+
+    /// Called by a worker after it has successfully executed the sequential commit hook for `txn_idx`.
+    ///
+    /// This method performs two main actions:
+    /// 1. Updates the `committed_marker` for `txn_idx` from `PendingCommitHook` to `Committed`.
+    ///    Panics if the previous marker was not `PendingCommitHook`.
+    /// 2. Pushes `txn_idx` to the `post_commit_processing_queue`, making it available for
+    ///    a `PostCommitProcessing` task to be dispatched by `[SchedulerV2::next_task]`.
+    ///    Panics if the queue push fails (e.g., if the queue is full, which shouldn't happen
+    ///    given it's bounded by `num_txns`).
+    ///
+    /// It is crucial that `txn_idx` was previously obtained from a successful call to
+    /// `[SchedulerV2::try_get_sequential_commit_hook]`, and that the `queueing_commits_lock` was held
+    /// by the worker during the execution of the commit hook and this call.
+    pub(crate) fn commit_hook_performed(&self, txn_idx: TxnIndex) -> Result<(), PanicError> {
+        // Allows next sequential commit hook to be processed.
+        if self.committed_marker[txn_idx as usize]
+            .swap(CommitMarkerFlag::Committed as u8, Ordering::Relaxed)
+            != CommitMarkerFlag::PendingCommitHook as u8
+        {
+            return Err(code_invariant_error(format!(
+                "Marking txn {} as COMMITTED, but previous marker != PENDING_COMMIT_HOOK",
+                txn_idx
+            )));
+        }
+
+        if self.post_commit_processing_queue.push(txn_idx).is_err() {
+            return Err(code_invariant_error(format!(
+                "Error adding {txn_idx} to commit queue, len {}",
+                self.post_commit_processing_queue.len()
+            )));
+        }
+
+        Ok(())
+    }
+
+    /// Unlocks the `queueing_commits_lock`.
+    ///
+    /// This method must be called by a worker after it has finished attempting to acquire
+    /// and process sequential commit hooks (i.e., after its loop calling
+    /// `[SchedulerV2::try_get_sequential_commit_hook]` and `[SchedulerV2::commit_hook_performed]` is done.
+    ///
+    /// If the next transaction to commit (`next_to_commit_idx`) is ready (i.e., executed
+    /// and not halted), this method will "arm" the lock, signifying that new commit work
+    /// might be available.
+    pub(crate) fn commit_hooks_unlock(&self) {
```

Comment:
Never called in this PR

Reply:
it's many APIs like this, since it's only scheduler so far. Some I removed from the original PR and will later with uses, but I tried to include the ones that can be somewhat explained with comments and tests.